pkg/operator/controller/gatewayapi/controller.go

```diff
@@ -51,6 +54,16 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(source.Kind(operatorCache, &configv1.FeatureGate{}), &handler.EnqueueRequestForObject{}, clusterNamePredicate); err != nil {
 		return nil, err
 	}
+
+	// watch for CRDs
+	if err = c.Watch(source.Kind(operatorCache, &apiextensionsv1.CustomResourceDefinition{}), &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return true },
```

Comment:
So this reconciles this control loop whenever there is any delete in an object informed by the `operatorCache`. I suppose that's not too bad, but generally, I think we try to be as precise as possible, to avoid the growing pains that come with unnecessary Operator reconcile cycles. Never know what crazy deletes/creates/updates a cluster at scale might be doing.

Have you thought about using the `managedCRDs` object and filtering the `DeleteEvent` by only when the CRDs we care about are deleted?

Reply:
@gcs278 The `managedCRDs` function already handles the case where if the CRDs are already present it will not do anything.
However I  added it for only delete events because that is what only required in this case, I don't want to add any extra events if they are not required at this moment.

Reply:
> @gcs278 The managedCRDs function already handles the case where if the CRDs are already present it will not do anything.

Correct, `ensureGatewayAPICRDs` won't do anything incorrectly if it gets extra reconciles, everything will work fine, but it's doing more work than it needs, and logging extra reconcile loops for other-than-gateway-api CRDs. e.g. when a `IngressController` is deleted in openshift-ingress-operator namespace, this logic will run reconcile, but we know deleting an `IngressController` won't cause any impact. 

I'm suggesting to be more specific from watching all CRDs delete events to only the managedCRDs we care about, maybe this suggestion will help to make sense (I haven't tested it):
```
	for i := range managedCRDs {
		if err = c.Watch(source.Kind(operatorCache, managedCRDs[i]), &handler.EnqueueRequestForObject{}, predicate.Funcs{
			CreateFunc:  func(e event.CreateEvent) bool { return false },
			DeleteFunc:  func(e event.DeleteEvent) bool { return true },
			UpdateFunc:  func(e event.UpdateEvent) bool { return false },
			GenericFunc: func(e event.GenericEvent) bool { return false },
		}); err != nil {
			return nil, err
		}
	}
```